### PR TITLE
Added functionality to remove attachment from a document by attachment name.

### PR DESCRIPTION
**Pull request by michaelwj**
Thanks for your hard work, please ensure all items are complete before opening.

- [x] You have signed the CLA as per the instructions in [CONTRIBUTING.md](https://github.com/cloudant/java-cloudant/blob/master/CONTRIBUTING.md#contributor-license-agreement)
- [X] You have added tests for any code changes
- [X] You have updated the [CHANGES.md](https://github.com/cloudant/java-cloudant/blob/master/CHANGES.md) 
- [X] You have completed the PR template below:

## What

Added functionality to cloudant-client to remove attachment from a document by attachment name.

## How

Needed ability to remove an attachment from a document by attachment name.

Modeled new functionality after the remove() methods that remove a document from the database. Leveraged functionality already available in the REST API to send a delete request with attachment name and _rev as query parameter (info found at https://docs.cloudant.com/attachments.html#delete).

## Testing

Added tests to com.cloudant.tests.AttachmentTest:

1) Creates a document, adds an attachment, verifies attachment has been added, and then calls the new removeAttachment method. Asserts the bar.getAttachments() is null after removal.

2) Test variant where id is null

3) Test variant where rev is null

4) Test variant where attachmentName is null

## Issues

N/A

### CODE REVIEW

**Comment by ricellis** [comment]
Thanks @michaelwj we'll get this reviewed ASAP.

**Comment by mikerhodes** [comment]
Just a note to confirm we have the CLA on file.

File: cloudant-client/src/main/java/com/cloudant/client/api/Database.java [hunk 1]
```diff
@@ -1010,6 +1010,59 @@ public boolean contains(String id) {
...
+     * String attachmentName = "example.jpg";
+     * //now remove the remote Bar
+     * Response response = db.removeAttachment(bar, attachmentName);
+     * }
+     * </pre>
+     *
+     * @param object the document to remove as an object
+     * @param attachmentName the attachment name to remove
+     * @return {@link com.cloudant.client.api.model.Response}
+     * @throws NoDocumentException If the document is not found in the database.
```
Comment (rhyshort) [inline cloudant-client/src/main/java/com/cloudant/client/api/Database.java]: I think `DocumentConflictException` should be included in throws list of the documentation.

File: cloudant-client/src/main/java/com/cloudant/client/api/Database.java [hunk 2]
```diff
@@ -1010,6 +1010,59 @@ public boolean contains(String id) {
...
+     * {@code
+     * Response response = db.removeAttachment("exampleId", "1-12345exampleRev", "example.jpg");
+     * }
+     * </pre>
+     *
+     * @param id  the document _id field
+     * @param rev the document _rev field
+     * @param attachmentName the attachment name
+     * @return {@link com.cloudant.client.api.model.Response}
+     * @throws DocumentConflictException if the attachment cannot be saved because of a conflict
```
Comment (rhyshort) [inline cloudant-client/src/main/java/com/cloudant/client/api/Database.java]: I think `NoDocumentException` should be in the throws list for the documentation.

File: cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java [hunk 3]
```diff
@@ -348,6 +348,41 @@ public Response saveAttachment(InputStream in, String name, String contentType,
     }
 
     /**
+     * Removes an attachment from a document.
+     * <p>The object must have the correct <code>_id</code> and <code>_rev</code> and <code>attachmentName</code> values.
+     *
+     * @param object The document to remove attachment as object.
+     * @param attachmentName The attachment name to remove.
+     * @return {@link Response}
+     * @throws NoDocumentException If the document is not found in the database.
```
Comment (rhyshort) [inline cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java]: I think `DocumentConflictException` should be in the throws list for the documentation.

File: cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java [hunk 4]
```diff
@@ -348,6 +348,41 @@ public Response saveAttachment(InputStream in, String name, String contentType,
...
+
+    /**
+     * Removes an attachment from a document given both a document <code>_id</code> and
+     * <code>_rev</code> and <code>attachmentName</code> values.
+     *
+     * @param id  The document _id field.
+     * @param rev The document _rev field.
+     * @param attachmentName The document attachment field.
+     * @return {@link Response}
+     * @throws DocumentConflictException
```
Comment (rhyshort) [inline cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDatabaseBase.java]: I think `NoDocumentException` should be in the throws list for the documentation.

**Comment by rhyshort** [comment]
@michaelwj it looks great, just a couple of minor documentation things and I think we could also do with some tests. Just to make sure that in the future we don't in inadvertently break the API. So I think we are looking at tests for both variants of `removeAttachment` where:

1. the `_id` field is `null`
2. the `rev` field is `null` 

Once that's done, we should be able to get it merged.

**Comment by michaelwj** [comment]
Thanks @rhyshort! I'll make the documentation changes and add the additional tests. Appreciate the feedback!

**Comment by emlaver** [comment]
The documentation changes and additional test cases look good to me +1

**Comment by rhyshort** [comment]
Thanks @michaelwj , looks good to merge, can you squash down into a single commit, and then I'll merge it in :)

**Comment by michaelwj** [comment]
Thanks @emlaver and @rhyshort, as requested, squashed to single commit. Again, appreciate the help!

**Comment by rhyshort** [comment]
Great, thanks for your hard work @michaelwj, merging.

**Pull request merged**